core-bundle/src/Resources/contao/widgets/Picker.php

```diff
@@ -0,0 +1,273 @@
+	 */
+	public function generate()
+	{
+		$blnHasOrder = ($this->orderField != '' && \is_array($this->{$this->orderField}));
+		$arrValues = $this->generateValues($blnHasOrder);
+		$arrSet = array_keys($arrValues);
+
+		$return = '<input type="hidden" name="'.$this->strName.'" id="ctrl_'.$this->strId.'" value="'.implode(',', $arrSet).'">' . ($blnHasOrder ? '
+  <input type="hidden" name="'.$this->strOrderName.'" id="ctrl_'.$this->strOrderId.'" value="'.$this->{$this->orderField}.'">' : '') . '
+  <div class="selector_container">' . (($blnHasOrder && \count($arrValues) > 1) ? '
+    <p class="sort_hint">' . $GLOBALS['TL_LANG']['MSC']['dragItemsHint'] . '</p>' : '') . '
+    <ul id="sort_'.$this->strId.'" class="'.($blnHasOrder ? 'sortable' : '').'">';
+
+		foreach ($arrValues as $k=>$v)
+		{
+			$return .= '<li data-id="'.$k.'">'.$v.'</li>';
+		}
+
+		$return .= '</ul>';
+
+		if (!System::getContainer()->get('contao.picker.builder')->supportsContext($this->context))
+		{
+			$return .= '
+	<p><button class="tl_submit" disabled>'.$GLOBALS['TL_LANG']['MSC']['changeSelection'].'</button></p>';
+		}
+		else
+		{
+			$extras = array
+			(
+				'fieldType' => $this->fieldType,
+				'source' => $this->strTable.'.'.$this->currentRecord,
+			);
+
+			$return .= '
+    <p><a href="' . ampersand(System::getContainer()->get('contao.picker.builder')->getUrl($this->context, $extras)) . '" class="tl_submit" id="picker_' . $this->strName . '">'.$GLOBALS['TL_LANG']['MSC']['changeSelection'].'</a></p>
+    <script>
+      $("picker_' . $this->strName . '").addEvent("click", function(e) {
+        e.preventDefault();
+        Backend.openModalSelector({
+          "id": "tl_listing",
+          "title": ' . json_encode($GLOBALS['TL_DCA'][$this->strTable]['fields'][$this->strField]['label'][0]) . ',
+          "url": this.href + document.getElementById("ctrl_' . $this->strId . '").value,
+          "callback": function(table, value) {
+            new Request.Contao({
+              evalScripts: false,
+              onSuccess: function(txt, json) {
+                $("ctrl_' . $this->strId . '").getParent("div").set("html", json.content);
+                json.javascript && Browser.exec(json.javascript);
+                $("ctrl_' . $this->strId . '").fireEvent("change");
+              }
+            }).post({"action":"reloadPicker", "name":"' . $this->strName . '", "value":value.join("\t"), "REQUEST_TOKEN":"' . REQUEST_TOKEN . '"});
+          }
+        });
+      });
+    </script>' . ($blnHasOrder ? '
+    <script>Backend.makeMultiSrcSortable("sort_'.$this->strId.'", "ctrl_'.$this->strOrderId.'", "ctrl_'.$this->strId.'")</script>' : '');
+		}
+
+		$return = '<div>' . $return . '</div></div>';
+
+		return $return;
+	}
+
+	protected function generateValues($blnHasOrder): array
+	{
+		$arrRelations = DcaExtractor::getInstance($this->strTable)->getRelations();
+		$strRelatedTable = $arrRelations[$this->strField]['table'] ?? null;
+
+		if (!$strRelatedTable) {
```

Comment:
CS

Reply:
Thanks, fixed in 1ce12911b7031b15d5436aa79e12b8dc87caedbe